blog/2024-07-07-openhab-4-2-release.md

```diff
@@ -0,0 +1,372 @@
+Please refer to the [**release notes**](https://github.com/openhab/openhab-distro/releases/tag/4.2.0) to check out what's new and noteworthy (e.g. breaking changes).
+
+In the following sections, our maintainers and contributors are introducing some new features you don't want to miss!
+
+## Mobile Notification Enhancements
+
+_Dan Cunningham ([@digitaldan](https://github.com/digitaldan)), openHAB Maintainer_
+
+With openHAB 4.2 we have updated our mobile push notifications architecture, bringing modern, cutting edge notification features to both our Android and iOS clients.
+
+First, we have consolidated both mobile platforms to use Google Firebase Messaging, which greatly improves the reliability of push notifications from our cloud service and reduces the amount of work needed to add new features in a cross-platform way.
+
+This change has allowed us to add an incredible amount of new features to our app in a relatively short amount of time.
+
+### Notification Display Enhancements
+
+Notifications now contain many more additional features, such as:
+
+- Title
+  - Replaces the default "openHAB" text with a user supplied version
+- Media Attachments
+  - Add images (or a short video on iOS) from an external URL or openHAB image item
+- On Click Action
+  - Navigate to a specific sitemap, page, popup widget, etc... in our UIs when the notification is clicked
+- Action Buttons
+  - Add up to 3 buttons that allow for executing commands, navigating through our UI's or open links in an embedded browser
+
+<p align="center"><img style="max-width: 70%;" src="/uploads/2024-07-07-openhab-4-2-release/notifications.png"/></p>
+
+### Update / Collapse Notifications
+
+We have also added some really useful ways to manage notifications already pushed to a device through rules and actions!
+
+- Collapsible Notifications
+  - Instead of seeing 100 messages that you left the garage door open, have a messages replace the existing shown one, so only the latest message is show on a user's device.
+  - Update existing messages with new information, add additional actions, media, etc.. without showing multiple notifications.
+
+<p align="center"><img style="max-width: 70%;" src="/uploads/2024-07-07-openhab-4-2-release/collapse-notifications.png"/></p>
+
+### Remove / Hide Notifications
+
+Finally, this has been a long requested feature by users, so we are very excited to be able to support removable notifications in this release!
+
+- Hide / Remove Notifications
+  - Is an alert no longer valid? Need to display something only temporarily?
+  - Using rules or actions, you can remove already pushed notifications from your devices.
+
+### Scripting Language Support
+
+These new notification enhancements are of course available across all our rule languages, but we have introduced additional helper APIs in our [JavaScript Scripting](/addons/automation/jsscripting/#cloud-notification-actions) and JRuby Scripting implementations to make this even easier to use.
```

Comment:
```suggestion
These new notification enhancements are of course available across all our rule languages, but we have introduced additional helper APIs in our [JavaScript Scripting](/addons/automation/jsscripting/#cloud-notification-actions) and [JRuby Scripting](https://openhab.github.io/openhab-jruby/main/OpenHAB/Core/Actions/Notification.html) implementations to make this even easier to use.
```

Reply:
Thanks for the link, I was already searching for a reference that I that provide yesterday, but thought I should recheck after I’ve seen a JRuby PR with the adjustments.